# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/rental/tests/target_ty_hack.rs
# Commit: 16350e08d34f
# Full Hash: 16350e08d34ff06c7f46745c6b160c7a7bda9140
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2019-12-06 04:43:57
# Regressor Bug: 1586855
# File Overlap Count: 6
# Description:
#   bug 1586855 - incorporate CRLite filters into cert_storage r=jcj,kjacobs
#   
#   This patch implements CRLite lookups for TLS server certificate revocation
#   information in telemetry-only mode. It adds a new preference
#   "security.pki.crlite_mode" to control the behavior of this feature. Setting
# ==============================================================================

diff -r 64fd40663930 -r 16350e08d34f third_party/rust/rental/tests/target_ty_hack.rs
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/third_party/rust/rental/tests/target_ty_hack.rs	Thu Dec 05 22:41:53 2019 +0000
@@ -0,0 +1,54 @@
+#[macro_use]
+extern crate rental;
+
+
+type MyVec<T> = Vec<T>;
+
+
+rental! {
+	pub mod rent_vec_slice {
+		use super::*;
+
+		#[rental]
+		pub struct OwnedSlice {
+			#[target_ty = "[u8]"]
+			buffer: MyVec<u8>,
+			slice: &'buffer [u8],
+		}
+
+		#[rental_mut]
+		pub struct OwnedMutSlice {
+			#[target_ty = "[u8]"]
+			buffer: MyVec<u8>,
+			slice: &'buffer mut [u8],
+		}
+	}
+}
+
+
+#[test]
+fn new() {
+	let vec = vec![1, 2, 3];
+	let _ = rent_vec_slice::OwnedSlice::new(vec, |slice| slice);
+}
+
+
+#[test]
+fn read() {
+	let vec = vec![1, 2, 3];
+	let rvec = rent_vec_slice::OwnedSlice::new(vec, |slice| slice);
+
+	assert_eq!(rvec.rent(|slice| slice[1]), 2);
+	assert_eq!(rvec.rent(|slice| slice[1]), rvec.rent(|slice| slice[1]));
+}
+
+
+#[test]
+fn write() {
+	let vec = vec![1, 2, 3];
+	let mut rvec = rent_vec_slice::OwnedMutSlice::new(vec, |slice| slice);
+
+	rvec.rent_mut(|slice| slice[1] = 4);
+	assert_eq!(rvec.rent(|slice| slice[1]), 4);
+	assert_eq!(rvec.rent(|slice| slice[1]), rvec.rent(|slice| slice[1]));
+}